tests_end_to_end/tests/Datasets/datasets_utils.py

```diff
@@ -0,0 +1,116 @@
+import time
+from opik import Opik
+import json
+from page_objects.DatasetsPage import DatasetsPage
+from page_objects.DatasetItemsPage import DatasetItemsPage
+from playwright.sync_api import Page
+import opik
+
+TEST_ITEMS = [
+    {'input': 'input0', 'output': 'output0'},
+    {'input': 'input1', 'output': 'output1'},
+    {'input': 'input2', 'output': 'output2'},
+    {'input': 'input3', 'output': 'output3'},
+    {'input': 'input4', 'output': 'output4'},
+    {'input': 'input5', 'output': 'output5'},
+    {'input': 'input6', 'output': 'output6'},
+    {'input': 'input7', 'output': 'output7'},
+    {'input': 'input8', 'output': 'output8'},
+    {'input': 'input9', 'output': 'output9'}
+]
+
+
+TEST_ITEMS_UPDATE = [
+    {'input': 'update-input0', 'output': 'update-output0'},
+    {'input': 'update-input1', 'output': 'update-output1'},
+    {'input': 'update-input2', 'output': 'update-output2'},
+    {'input': 'update-input3', 'output': 'update-output3'},
+    {'input': 'update-input4', 'output': 'update-output4'},
+    {'input': 'update-input5', 'output': 'update-output5'},
+    {'input': 'update-input6', 'output': 'update-output6'},
+    {'input': 'update-input7', 'output': 'update-output7'},
+    {'input': 'update-input8', 'output': 'update-output8'},
+    {'input': 'update-input9', 'output': 'update-output9'}
+]
+
+
+def insert_dataset_items_sdk(client: opik.Opik, dataset_name, items_list):
+    dataset = client.get_dataset(dataset_name)
+    dataset.insert(items_list)
+
+
+def insert_dataset_items_ui(page: Page, dataset_name, items_list):
+    datasets_page = DatasetsPage(page)
+    datasets_page.go_to_page()
+    datasets_page.select_database_by_name(dataset_name)
+
+    dataset_items_page = DatasetItemsPage(page)
+    
+    for item in items_list:
+        dataset_items_page.insert_dataset_item(json.dumps(item))
+
+
+def delete_one_dataset_item_sdk(client: opik.Opik, dataset_name):
+    dataset = client.get_dataset(dataset_name)
+    item = dataset.get_items()[0]
+    dataset.delete([item['id']])
+    return item
+
+
+def delete_one_dataset_item_ui(page: Page, dataset_name):
+    dataset_page = DatasetsPage(page)
+    dataset_page.go_to_page()
+    dataset_page.select_database_by_name(dataset_name)
+    dataset_items_page = DatasetItemsPage(page)
+    item = dataset_items_page.delete_first_item_on_page_and_return_content()
+    return item
+
+
+def wait_for_dataset_to_be_visible(client: Opik, dataset_name: str, timeout=10):
+    start_time = time.time()
+    dataset = None
+    while time.time() - start_time < timeout:
+        dataset = None
```

Comment:
Lets put a small sleep ( e.g. 0.1 seconds) after every attempt
Btw we have quite similar utility in `opik.synchronization`